src/encoder.js

```diff
@@ -45,10 +45,22 @@ function encoder(mtype) {
 
         // Map fields
         if (field.map) {
+            let keyGetter;
+            switch (field.keyType) {
+                case "int64":
```

Comment:
Probably also makes sense to extend this for `uint64`?

Reply:
I think you are right. Likely also for `sint64`, `fixed64`, `sfixed64` as well. Would you like me to extend the PR with these?

Reply:
That would be amazing, yeah! Thanks so much! Do you think we'd see this merged to `master` at some point? (Cc: @dcodeIO)

Reply:
My pleasure! :) I've added this.

Imo it should be merged to master, as it is definitely broken without this fix.